### PR TITLE
Support specific launch activity name for Android

### LINKED ISSUE #935: Activity class does not exists
## Actual Behavior
1. Execute "React Native: Run Android on Simulator" from palette
2. App is built and installed successfully
3. App is not launched because we're adding an applicationIdSuffix ".development" in buildType debug from gradle and our launch activity is called HomeActivity rather than MainActivity

## Expected Behavior
1. We'd like to specify packageName and activity to launch in settings

## Software versions
- React-Native VS Code extension version: 0.8.0
- VSCode version: 1.32.
- OS platform and version:  macOS Mojave 10.14.3
- NodeJS version: 10.15.3
- React Native version: 0.58.4
- Expo SDK version (include if relevant):

## Outputs (Include if relevant)
- Output of the Debug Console (View -> Toggle Debug Console):
```
BUILD SUCCESSFUL in 12s
30 actionable tasks: 6 executed, 24 up-to-date
Running /home/xpuser/Android/Sdk/platform-tools/adb -s emulator-5554 reverse tcp:8081 tcp:8081
Starting the app on emulator-5554 (/home/xpuser/Android/Sdk/platform-tools/adb -s emulator-5554 shell am start -n it.trovaprezzi.android/it.trovaprezzi.android.MainActivity)...
Starting: Intent { cmp=it.trovaprezzi.android/.MainActivity }
Error type 3
Error: Activity class {it.trovaprezzi.android/it.trovaprezzi.android.MainActivity} does not exist.
```

### PR DESCRIPTION
**Pull request by RedMickey**
Added the opportunity to specify the name of Android launch activity.
Resolves #935

### CODE REVIEW

File: src/extension/android/androidPlatform.ts [hunk 1]
```diff
@@ -90,6 +91,15 @@ export class AndroidPlatform extends GeneralMobilePlatform {
                         this.runArguments.push("--no-packager");
                     }
 
+                    let mainActivity = GeneralMobilePlatform.getOptFromRunArgs(this.runArguments, "--main-activity");
+
+                    if (mainActivity && mainActivity !== "") {
```
Comment (SounD120) [inline src/extension/android/androidPlatform.ts]:
```suggestion
                    if (mainActivity) {
```

File: src/extension/settingsHelper.ts [hunk 2]
```diff
@@ -58,11 +59,24 @@ export class SettingsHelper {
...
-            return ConfigurationReader.readArray(workspaceConfiguration.get(configKey));
+            runArgs.push(...ConfigurationReader.readArray(workspaceConfiguration.get(configKey)));
+        }
+        if (
+            !GeneralMobilePlatform.getOptFromRunArgs(runArgs, "--main-activity", true)
+            && workspaceConfiguration.has(launchActivityKey)
+            && platform === "android"
+          ) {
+            const launchActivity: string = ConfigurationReader.readString(workspaceConfiguration.get(launchActivityKey));
+            if (launchActivity !== "") {
```
Comment (SounD120) [inline src/extension/settingsHelper.ts]:
```suggestion
            if (launchActivity) {
```

File: package.json [hunk 3]
```diff
@@ -233,6 +233,11 @@
 								"type": "string",
 								"description": "%reactNative.attach.debuggerWorkerUrlPath.description%",
 								"default": "debugger-ui/"
+							},
+							"launchActivity": {
```
Comment (SounD120) [inline package.json]: Please make sure that we need this property for attach scenario.

File: package.nls.json [hunk 4]
```diff
@@ -28,6 +28,7 @@
     "reactNative.attach.localRoot.description":"The local source root that corresponds to the 'remoteRoot'.",
     "reactNative.attach.skipFiles.description":"An array of file or folder names, or glob patterns, to skip when debugging.",
     "reactNative.attach.debuggerWorkerUrlPath.description": "Path to the app debugger worker to override. For example, if debugger tries to attach to http://localhost:8081/debugger-ui/debuggerWorker.js and you get 404 error from packager output then you may want to change debuggerWorkerUrlPath to another value suitable for your packager (\"debugger-ui\" will be replaced with the value you provide).",
+    "reactNative.attach.launchActivity.description": "The activity to be attached for debugging. Default is 'MainActivity'.",
```
Comment (SounD120) [inline package.nls.json]: Please make sure that we need this property for attach scenario.

**Comment by SounD120** [comment]
@RedMickey Please add a description for `launchActivity` property to the [debug configurations properties list](https://github.com/microsoft/vscode-react-native#react-native-debug-configuration-properties). And also it would be great if new `settings.json` parameter `launchActivity` will be documented [here](https://github.com/microsoft/vscode-react-native/blob/master/doc/customization.md). Thank you.

File: README.md [hunk 5]
```diff
@@ -63,6 +63,7 @@ See [Setting up debug environment](doc/debugging.md) for more details.
 |`target`|Target to run on. Possible values: `simulator`, `device`, `<Android emulator/device id>`, `<iOS simulator/device name>`|`string`|`simulator`|
 |`logCatArguments`|Arguments to be used for LogCat (The LogCat output will appear on an Output Channel). It can be an array such as: `[":S", "ReactNative:V", "ReactNativeJS:V"]`|`array`|`["*:S", "ReactNative:V", "ReactNativeJS:V"]`|
 |`runArguments`|Run arguments to be passed to `react-native run-<platform>` command (override all other configuration params)|`array`|n/a|
+|`launchActivity`|The activity to be launched for debugging, specifies `--main-activity` parameter in run arguments|`string`|`MainActivity`|
```
Comment (SounD120) [inline README.md]:
```suggestion
|`launchActivity`|The Android activity to be launched for debugging, specifies `[--main-activity](https://github.com/react-native-community/cli/blob/master/docs/commands.md#--main-activity-string)` parameter in `react-native` run arguments|`string`|`MainActivity`|
```

File: README.md [hunk 6]
```diff
@@ -63,6 +63,7 @@ See [Setting up debug environment](doc/debugging.md) for more details.
 |`target`|Target to run on. Possible values: `simulator`, `device`, `<Android emulator/device id>`, `<iOS simulator/device name>`|`string`|`simulator`|
 |`logCatArguments`|Arguments to be used for LogCat (The LogCat output will appear on an Output Channel). It can be an array such as: `[":S", "ReactNative:V", "ReactNativeJS:V"]`|`array`|`["*:S", "ReactNative:V", "ReactNativeJS:V"]`|
 |`runArguments`|Run arguments to be passed to `react-native run-<platform>` command (override all other configuration params)|`array`|n/a|
+|`launchActivity`|The Android activity to be launched for debugging, specifies [`--main-activity`](https://github.com/react-native-community/cli/blob/master/docs/commands.md#--main-activity-string) parameter in `react-native` run arguments|`string`|`MainActivity`|
```
Comment (ruslan-bikkinin) [inline README.md]:
```suggestion
|`launchActivity`|The Android activity to be launched for debugging, e.g. it specifies [`--main-activity`](https://github.com/react-native-community/cli/blob/master/docs/commands.md#--main-activity-string) parameter in `react-native` run arguments|`string`|`MainActivity`|
```

File: src/extension/settingsHelper.ts [hunk 7]
```diff
@@ -84,4 +84,4 @@ export class SettingsHelper {
 
         return "";
     }
-}
```
Comment (ruslan-bikkinin) [inline src/extension/settingsHelper.ts]: Please, remove extra new line (redundant change)

**Pull request merged**